### ISSUE TITLE
simple_broadcast test showing wrong values

### ISSUE THREAD

**Issue by svuckovicTT** [tenstorrent/tt-mlir#861]
Tested on 8c6494cb.

Running:
```bash
ttrt query --save-artifacts
export SYSTEM_DESC_PATH=ttrt-artifacts/system_desc.ttsys
./build/bin/ttmlir-opt --ttir-to-ttnn-backend-pipeline="system-desc-path=ttrt-artifacts/system_desc.ttsys" test/ttmlir/Silicon/TTNN/simple_broadcast.mlir > binary.mlir
./build/bin/ttmlir-translate --ttnn-to-flatbuffer binary.mlir -o binary.ttnn
ttrt run binary.ttnn
```

produces
```bash
...
2024-10-07 12:13:00,426 - DEBUG - input tensors for program=0
2024-10-07 12:13:00,426 - DEBUG - output tensors for program=0
2024-10-07 12:13:00,427 - DEBUG - tensor([0.3909, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
        0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
        0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
        0.0000, 0.0000, 0.0000, 0.0000, 0.0000])
...
```

Two things are unclear:
1. first value of tensor is non-zero, but a 0 constant should be broadcasted as defined in the test
2. all the other values are zeros (which should be correct if the first value is zero but isn't)

Ideally, the test should broadcast a non-zero value...

**Comment by tapspatel** [comment]
@uazizTT this seems like some runtime error, let me know if it's in your domain. Else we can reassign to runtime folks.

**Comment by uazizTT** [comment]
@tapspatel Yes I agree, please reassign to runtime team.

**Comment by tapspatel** [comment]
@jnie-TT this seems like the issue we were talking about offline, let's discuss how we can support 0 values + some type of golden that's not zero-check

**Comment by jnie-TT** [comment]
@svuckovicTT @uazizTT it seems the issue is the ttnnir is ill-formed:
```
#device = #tt.device<workerGrid = #tt.grid<7x8, (d0, d1) -> (0, d0, d1)>, l1Map = (d0, d1)[s0, s1] -> (0, d0 floordiv s0, d1 floordiv s1, (d0 mod s0) * s1 + d1 mod s1), dramMap = (d0, d1)[s0, s1] -> (0, 0, ((((d0 floordiv s0) * 8 + d1 floordiv s1) * (s1 * s0) + (d0 mod s0) * s1 + d1 mod s1) floordiv 8192) mod 12, (((d0 floordiv s0) * 8 + d1 floordiv s1) * (s1 * s0) + (d0 mod s0) * s1 + d1 mod s1) floordiv 98304 + (((d0 floordiv s0) * 8 + d1 floordiv s1) * (s1 * s0) + (d0 mod s0) * s1 + d1 mod s1) mod 8192), meshShape = , chipIds = [0]>
#dram = #tt.memory_space<dram>
#system = #tt.memory_space<system>
#system_desc = #tt.system_desc<[{arch = <wormhole_b0>, grid = 7x8, l1_size = 1499136, num_dram_channels = 12, dram_channel_size = 1073741824, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32, l1_unreserved_base = 107616, erisc_l1_unreserved_base = 107616, dram_unreserved_base = 32, dram_unreserved_end = 1073149920, physical_cores = {worker = [ 1x1,  1x2,  1x3,  1x4,  1x6,  1x7,  1x8,  1x9,  2x1,  2x2,  2x3,  2x4,  2x6,  2x7,  2x8,  2x9,  3x1,  3x2,  3x3,  3x4,  3x6,  3x7,  3x8,  3x9,  4x1,  4x2,  4x3,  4x4,  4x6,  4x7,  4x8,  4x9,  7x1,  7x2,  7x3,  7x4,  7x6,  7x7,  7x8,  7x9,  8x1,  8x2,  8x3,  8x4,  8x6,  8x7,  8x8,  8x9,  9x1,  9x2,  9x3,  9x4,  9x6,  9x7,  9x8,  9x9] dram = [ 1x0,  1x5,  2x5,  3x5,  5x0,  5x5,  7x0,  7x5,  8x5,  9x5,  11x0,  11x5] eth = [ 6x9] eth_inactive = [ 0x1,  0x2,  0x3,  0x4,  0x6,  0x7,  0x8,  0x9,  6x2,  6x3,  6x6,  6x7,  6x8]}, supported_data_types = [<f32>, <f16>, <bf16>, <bfp_f8>, <bfp_bf8>, <bfp_f4>, <bfp_bf4>, <bfp_f2>, <bfp_bf2>, <u32>, <u16>, <u8>], supported_tile_sizes = [ 4x16,  16x16,  32x16,  4x32,  16x32,  32x32]}, {arch = <wormhole_b0>, grid = 7x8, l1_size = 1499136, num_dram_channels = 12, dram_channel_size = 1073741824, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32, l1_unreserved_base = 107616, erisc_l1_unreserved_base = 107616, dram_unreserved_base = 32, dram_unreserved_end = 1073149920, physical_cores = {worker = [ 1x1,  1x2,  1x3,  1x4,  1x6,  1x7,  1x8,  1x9,  3x1,  3x2,  3x3,  3x4,  3x6,  3x7,  3x8,  3x9,  4x1,  4x2,  4x3,  4x4,  4x6,  4x7,  4x8,  4x9,  5x1,  5x2,  5x3,  5x4,  5x6,  5x7,  5x8,  5x9,  7x1,  7x2,  7x3,  7x4,  7x6,  7x7,  7x8,  7x9,  8x1,  8x2,  8x3,  8x4,  8x6,  8x7,  8x8,  8x9,  9x1,  9x2,  9x3,  9x4,  9x6,  9x7,  9x8,  9x9] dram = [ 1x0,  1x5,  2x5,  3x5,  5x0,  5x5,  7x0,  7x5,  8x5,  9x5,  11x0,  11x5] eth = [ 0x9] eth_inactive = [ 0x2,  0x3,  0x4,  0x6,  0x7,  0x8,  6x1,  6x2,  6x3,  6x4,  6x6,  6x7,  6x8,  6x9]}, supported_data_types = [<f32>, <f16>, <bf16>, <bfp_f8>, <bfp_bf8>, <bfp_f4>, <bfp_bf4>, <bfp_f2>, <bfp_bf2>, <u32>, <u16>, <u8>], supported_tile_sizes = [ 4x16,  16x16,  32x16,  4x32,  16x32,  32x32]}], [0, 1], [3 : i32, 0 : i32], [ 0x0x0x0], [<[0, 8, 0], [1, 0, 0]>]>
#layout = #tt.layout<(d0) -> (0, d0), undef, <1x1>, memref<1x32xf32, #system>>
#layout1 = #tt.layout<(d0) -> (0, d0), undef, <1x1>, memref<1x1xf32, #dram>, interleaved>
#layout2 = #tt.layout<(d0) -> (0, d0), undef, <1x1>, memref<1x32xf32, #dram>, interleaved>
module attributes {tt.device = #device, tt.system_desc = #system_desc} {
  func.func public @broadcast() -> tensor<32xf32, #layout> {
    %0 = "ttnn.get_device"() <{mesh_shape = #ttnn<mesh_shape 1x1>}> : () -> !tt.device<#device>
    %1 = "ttnn.empty"(%0) <{dtype = #tt.supportedDataTypes<f32>, layout = #ttnn.layout<row_major>, memory_config = #ttnn.memory_config<<interleaved>, <dram>>, shape = #ttnn.shape<1>}> : (!tt.device<#device>) -> tensor<1xf32, #layout1>
    %2 = "ttnn.empty"(%0) <{dtype = #tt.supportedDataTypes<f32>, layout = #ttnn.layout<row_major>, memory_config = #ttnn.memory_config<<interleaved>, <dram>>, shape = #ttnn.shape<32>}> : (!tt.device<#device>) -> tensor<32xf32, #layout2>
    %3 = "ttnn.empty"(%0) <{dtype = #tt.supportedDataTypes<f32>, layout = #ttnn.layout<row_major>, memory_config = #ttnn.memory_config<<interleaved>, <dram>>, shape = #ttnn.shape<32>}> : (!tt.device<#device>) -> tensor<32xf32, #layout2>
    %4 = "ttnn.empty"() <{dtype = #tt.supportedDataTypes<f32>, layout = #ttnn.layout<row_major>, shape = #ttnn.shape<32>}> : () -> tensor<32xf32, #layout>
    %5 = "ttnn.to_memory_config"(%1, %0) : (tensor<1xf32, #layout1>, !tt.device<#device>) -> tensor<32xf32, #layout>
    return %5 : tensor<32xf32, #layout>
  }
}
```
We're returning a tensor %5, which has 32 elements, which is the output of a to_memory_config, which takes tensor %1, which has only 1 element? Are we expecting the broadcast to happen in to_memory_config? Honestly I don't know what this test is doing, but it doesn't seem correct.

Probably related to this? https://github.com/tenstorrent/tt-mlir/commit/158b41a3752811c6f83b927448facfdf5bc4a86e

**Comment by svuckovicTT** [comment]
Hey @uazizTT, any thoughts on this?

**Comment by uazizTT** [comment]
> Hey @uazizTT, any thoughts on this?

It is related to removing the dead empty Op, should be fixed when the following PR is merged:
https://github.com/tenstorrent/tt-mlir/pull/882

**Comment by svuckovicTT** [comment]
Hey @uazizTT, I see that you've merged the change. Can you confirm this now works, and close the issue?

**Comment by uazizTT** [comment]
Yes this should be fixed with #882
